### PR TITLE
Player Database, Presence Kick Blocking, Net Messages, and More

### LINKED ISSUE #62: Add player database/block join
People suck. The block join feature shouldn't need to exist BUT...

### PR DESCRIPTION
**Pull request by Rxann**
closes #62 

thanks to @tyackman and @maybegreat48 for all your help on this

TODO:

| Task | Completed |
|--------|--------|
| Get ReceiveNetMessage to work | ✅ |
| Move InFrame to RDR-Classes| ✅ |
| Debug InFrame | ✅ |

### CODE REVIEW

File: src/game/hooks/Hooks.hpp [hunk 1]
```diff
@@ -11,6 +11,35 @@ namespace rage
 	class datBitBuffer;
 	class netObject;
 	class netSyncTree;
+	class CJoinRequestContext;
+	class CMsgJoinResponse;
+	class rlGamerInfo;
+	class CPlayerCameraDataNode;
+	class netConnectionManager;
+#pragma pack(push, 1)
```
Comment (Rxann) [inline src/game/hooks/Hooks.hpp:19]: Move to RDR-Classes

**Comment by AAA-ALR** [comment]
`SendPacket = ptr.Add(10).Rip().As<Functions::SendPacket>();`
Add(10) does not go to the function call address

SendPacket = ptr.Add(14).Add(1).Rip().As<Functions::SendPacket>();

**Comment by Rxann** [comment]
> `SendPacket = ptr.Add(10).Rip().As<Functions::SendPacket>();` Add(10) does not go to the function call address
> 
> SendPacket = ptr.Add(14).Add(1).Rip().AsFunctions::SendPacket();

Thank you, I seemingly overlooked this, idk how. but thank you for pointing it out

**Comment by AAA-ALR** [comment]
And the queue_packet function should have 7 parameters
using queue_packet = bool (*)(rage::netConnectionManager* mgr, int msg_id,int a3, void* data, int size, int flags, void* unk);

image

**Comment by Rxann** [comment]
> And the queue_packet function should have 7 parameters `using QueuePacket = bool (*)(rage::netConnectionManager* mgr, int msg_id,int a3, void* data, int size, int flags, void* unk);`
> 
> image

Well dang I missed some stuff. Thanks again for this. Any clue what value a3 normally is?

**Pull request merged**